From d75b9cbbe0c194e386ecc88129c0980e9b93ad69 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 02 Jan 2020 17:17:35 +0200
Subject: [PATCH] [SW-6739] arcboot: fix alignment caused by length

length is including it self in image size, calculate
position of signature and data taking this into account

Change-Id: Ic4b4f41f9d3c452c2ab3a517f64079c583ed2b08
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/arcboot_rom.c b/common/arcboot_rom.c
index a8e14bc..e010359 100644
--- a/common/arcboot_rom.c
+++ b/common/arcboot_rom.c
@@ -66,10 +66,10 @@
 {
 	int st, err;
 	unsigned char block_buf[512];
-	uint16_t test_phase0b_data_len = image_size - SIG_SIZE;
+	uint16_t test_phase0b_data_len = image_size - SIG_SIZE - BLOB_START;
 
 	unsigned char *test_phase0b_data = &fw_sram[BLOB_START];
-	unsigned char *test_phase0b_sig = &fw_sram[test_phase0b_data_len];
+	unsigned char *test_phase0b_sig = &fw_sram[test_phase0b_data_len + BLOB_START];
 	unsigned char *test_public_key = test_phase0b_sig - KEY_SIZE;
 	uint32_t test_phase0b_sig_len = SIG_SIZE;
 	uint32_t test_public_key_len = KEY_SIZE;
